lightning/src/chain/keysinterface.rs

```diff
@@ -394,6 +473,45 @@ impl ChannelKeys for InMemoryChannelKeys {
 		local_commitment_tx.get_htlc_sigs(&self.htlc_base_key, local_csv, secp_ctx)
 	}
 
+	fn sign_justice_transaction<T: secp256k1::Signing + secp256k1::Verification>(&self, justice_tx: &Transaction, input: usize, amount: u64, per_commitment_key: &SecretKey, htlc: &Option<HTLCOutputInCommitment>, on_remote_tx_csv: u16, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()> {
```

Comment:
The nesting here, as in the method below, is pretty deep. Could this be changed to either using `?` with corresponding From<Err> for () implementations, or using consecutive (non-nested) match clauses for the Options that either return the value to a top-level-scope variable or return from the method?

Reply:
Good idea, took the consecutive (non-nested) match clauses option.